packages/core/test/state.test.ts

```diff
@@ -825,13 +825,13 @@ describe('State', () => {
       expect(state.can({ type: 'ANY_EVENT' })).toEqual(false);
     });
 
-    it('should allow errors to propagate', () => {
+    it('should not execute assignments', () => {
```

Comment:
based on this test we can't truly know if it tests what the test title suggests - the error could be just swallowed somewhere within our implementation and the test would pass. Could we refactor this to use a variable that changes value within `assign` and assert that it has not been changed?

Reply:
Done